src/events/rawGuildMemberAdd.ts

```diff
@@ -6,25 +6,26 @@ import { Guild, GuildMember, MessageEmbed, Permissions, TextChannel, User } from
 import { Event, EventStore } from 'klasa';
 import { Colors } from '@lib/types/constants/Constants';
 import { floatPromise, resolveOnErrorCodes } from '@utils/util';
+import { LanguageKeysSimple } from '@lib/types/Augments';
 
 const { FLAGS } = Permissions;
-const REGEXP = /%MEMBER%|%MEMBERNAME%|%MEMBERTAG%|%GUILD%|%POSITION%|%MEMBERCOUNT%/g;
-const MATCHES = {
-	GUILD: '%GUILD%',
-	MEMBER: '%MEMBER%',
-	MEMBERNAME: '%MEMBERNAME%',
-	MEMBERTAG: '%MEMBERTAG%',
-	MEMBERCOUNT: '%MEMBERCOUNT%',
-	POSITION: '%POSITION%'
-};
-
-const COLORS = {
-	JOIN: { color: Colors.Green, title: 'Member Join' },
-	MUTE: { color: Colors.Amber, title: 'Muted Member Join' }
-};
+const enum Matches {
+	Guild = '%GUILD%',
+	Member = '%MEMBER%',
+	MemberName = '%MEMBERNAME%',
+	MemberTag = '%MEMBERTAG%',
+	MemberCount = '%MEMBERCOUNT%',
+	Position = '%POSITION%'
+}
 
 export default class extends Event {
 
+	private readonly kTransformMessageRegExp = /%MEMBER%|%MEMBERNAME%|%MEMBERTAG%|%GUILD%|%POSITION%|%MEMBERCOUNT%/g;
+	private readonly kMessageLogMetaData = {
+		Join: { color: Colors.Green, title: 'EVENTS_GUILDMEMBERADD' },
+		Mute: { color: Colors.Amber, title: 'EVENTS_GUILDMEMBERADD_MUTE' }
+	} as Record<string, MessageLogMetaData>;
```

Comment:
Mayhaps? Otherwise it's fine

```suggestion
	} as const;
```